### DISCUSSION TITLE
Redefinition main JS chunk (entry)

### DISCUSSION THREAD

**Discussion by iWatchYouFromAfar** [nuxt/nuxt#23433] [Questions]
### Describe the feature

Is there any way that i can override the position of the main JS chunk (entry.hash.js, which contains the main application code)? Or add defer/async attributes?

If there is no such possibility, i think it would be a good functionality for the flexibility of the Nuxt 3.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [X] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
Would you explain a bit more about what you're trying to do, and what you mean by 'override the position of the main JS chunk'?

**Reply by iWatchYouFromAfar** [reply]
I have an almost empty application (3 components, layout, a logo component, and a navbar component with menu items), there are no more plugins, modules or JS code and etc. My app has a good show of performance in Lighthouse - 98. However, the LCP score is 2.3-2.4 (at the same time, the LCP element is just a small text and fonts I have loaded with the preload attribute). To improve the LCP indicator, one of the possible solutions is to transfer the connection of the JS file from head tag, to before closing the body tag or add the defer attribute to the script tag.

I would like to try to implement this on an empty application and check if it affects the LCP. As i understand, for now there is no any way to manage JS chunk connection .

**Comment by iWatchYouFromAfar** [comment]
We found out that when specifying `type="module"`, you do not need to specify the `defer` attribute. Then it is not clear what can cause LCP 2.3-2.4 if there are only 2-3 components in the application...

**Reply by danielroe** [reply]
it could be a lot of things, but difficult to say without any code or example. maybe custom web fonts? is it server rendered or client?

**Reply by iWatchYouFromAfar** [reply]
SSR and only 2 custom fonts with a total size of 44 kb are used. I tried to disable them - there are not much changes.

To be clear, i create a repository in which i installed pure Nuxt 3 - https://github.com/iWatchYouFromAfar/test-nuxt3-lcp.

Added 1 module and slightly modified `nuxt.config.ts`. I didn't add components and CSS. That's what i have:

- 1 layout;
- 2 custom fonts with preload on SSR;
- all css (which i plan to add for testing later) inlined by hooks from Nuxt docs

And that's it.

Further

`npm run build && npm run preview` and [i get the following result](https://googlechrome.github.io/lighthouse/scorecalc/#FCP=1355&LCP=2034&TBT=0&CLS=0&SI=2885&TTI=1355&FMP=1355&device=mobile&version=10.2.0). 

[And these result are from my current Nuxt application](https://googlechrome.github.io/lighthouse/scorecalc/#FCP=1506&LCP=2413&TBT=0&CLS=0&SI=1506&TTI=1506&FMP=1506&device=mobile&version=10.2.0). 

As you can see, the LCP is 0.4 seconds longer. But i am interested in why on an empty Nuxt application, LCP immediately gets a 2.0 rating, just as FCP also has a significantly higher rating.

**Reply by danielroe** [reply]
This is likely due to your webfonts.

Try using https://github.com/nuxt-modules/fontaine.

We will be implementing some of these strategies in the upcoming nuxt/fonts.

**Reply by iWatchYouFromAfar** [reply]
I use it already in my test app - https://github.com/iWatchYouFromAfar/test-nuxt3-lcp.

**Comment by iWatchYouFromAfar** [comment]
After some tests, i noticed that my LCP increased by 0.3-0.4 seconds after I added the component with the logo:

`
<AElementsLogo>
  <a href="/">
    <picture>
      <source
        media="(min-width: 1200px)"
        srcset="~/assets/images/logo-desktop.svg"
      />
      <source
        media="(min-width: 576px)"
        srcset="~/assets/images/logo-mobile.svg"
      />
      <img src="~/assets/images/logo-mobile.svg" alt="Logotype" />
    </picture>
  </a>
</AElementsLogo>
`
